src/modules/sdlog2/sdlog2.c

```diff
@@ -1470,9 +1511,19 @@ int sdlog2_thread_main(int argc, char *argv[])
 		}
 
 		/* --- EKF2 REPLAY --- */
-		if(record_replay_log) {
-			// we poll on the replay topic so we know that it was updated
-			// but we need to copy it again since we are re-using the buffer.
+		if (log_type == LOG_TYPE_ALL || LOG_TYPE_REPLAY_ONLY) {
+
+			if (log_type == LOG_TYPE_ALL) {
+				// When logging everything we are polling for sensor_combined, so
+				// we need to use the usual copy_if_updated which includes orb_subscribe.
+				copy_if_updated(ORB_ID(ekf2_replay), &subs.replay_sub, &buf.replay);
+
+			} else {
+				// We poll on the replay topic so we know that it was updated
+				// but we need to copy it again since we are re-using the buffer.
+				orb_copy(ORB_ID(ekf2_replay), subs.replay_sub, &buf.replay);
+			}
+
 			orb_copy(ORB_ID(ekf2_replay), subs.replay_sub, &buf.replay);
```

Comment:
Yes, but if you remove it right now you'll end up with a busy loop on poll. Don't try to be too smart here. I've fixed this now three times because people didn't understand how poll works, I don't want to fix it a fourth time.